app.py

```diff
@@ -0,0 +1,139 @@
+from flask import Flask, request, jsonify, make_response
+from flask_sqlalchemy import SQLAlchemy
+from os import environ
+from abc import ABC, abstractmethod
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = environ.get('DATABASE_URL')
+db = SQLAlchemy(app)
+
+
+class User(db.Model):
+    __tablename__ = 'users'
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(30), unique=False, nullable=False)
+    email = db.Column(db.String(50), unique=True, nullable=False)
+
+    def json(self):
+        return {
+            'id': self.id,
+            'username': self.username,
+            'email': self.email
+        }
+
+
+class AbstractDatabaseAdapter(ABC):
+
+    @abstractmethod
+    def create(self, entity):
+        pass
+
+    @abstractmethod
+    def get_all(self, entity_type):
+        pass
+
+    @abstractmethod
+    def get_by_id(self, entity_type, entity_id):
+        pass
+
+    @abstractmethod
+    def update(self, entity):
+        pass
+
+    @abstractmethod
+    def delete(self, entity):
+        pass
+
+
+class PostgreSQLAdapter(AbstractDatabaseAdapter):
```

Comment:
In current solution this is not really an adapter, because it inherits from ABC which doesn't have any base behaviour. I know it's kinda hard to create an adapter out of nothing, but maybe you can try to find inspiration in this example: https://refactoring.guru/design-patterns/adapter/python/example#example-1 (you can try composition but inheritance is also fine if you're more comfortable with it)

Reply:
topic to be discussed at the meeting